scipy/optimize/tests/test_minimize_constrained.py

```diff
@@ -615,6 +615,32 @@ def callback(x, info):
         # compatibility
         assert_(result.get('niter', -1) == 1)
 
+    def test_issue_15093(self):
+        # scipy docs define bounds as inclusive, so it shouldn't be
+        # an issue to set x0 on the bounds even if keep_feasible is
+        # True. Previously, trust-constr would treat bounds as
+        # exclusive.
+
+        x0 = np.array([0., 0.5])
+
+        def obj(x):
+            x1 = x[0]
+            x2 = x[1]
+            return x1 ** 2 + x2 ** 2
+
+        bounds = Bounds(np.array([0., 0.]), np.array([1., 1.]),
+                        keep_feasible=True)
+
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning, "delta_grad == 0.0")
+            result = minimize(
+                method='trust-constr',
+                fun=obj,
+                x0=x0,
+                bounds=bounds)
+
+        assert_(result.get('success'))
```

Comment:
Minor: we prefer plain `assert` in *new* test cases.

More detailed:

I checked that this fails on `main` and passes here with `main` failure summarized below:

`scipy/optimize/tests/test_minimize_constrained.py::TestTrustRegionConstr::test_issue_15093 - RuntimeWarning: invalid value encountered in scalar subtract`

The original issue reported a hang with a different reproducer, but I can longer reproducer that hang on `main`--can you clarify what is going on there?

If I take the original reproducer, and set `keep_feasible=True`, it fails on this branch as well--is that expected? The failure is different than on `main`, but still hard for me to judge.

Reply:
Having just examined the test problem I'm surprised that `result.x` doesn't arrive very close to the true solution (it's something like 2e-3 off zero). I wonder if the default gtol could be lowered.